test-kotlin/src/com/intellij/rt/coverage/runner.kt

```diff
@@ -43,3 +46,14 @@ private fun statusToString(lines: Map<Int, Byte>) = lines.mapValues {
 }
 
 private fun ClassData.getLinesData() = lines.filterIsInstance(LineData::class.java).sortedBy { it.lineNumber }
+
+private val coverageMarkerRegex = Regex("// coverage: (FULL|PARTIAL|NONE)( .*)?\$")
+
+internal fun extractCoverageDataFromFile(file: File): Map<Int, String> = file.bufferedReader()
+        .lineSequence()
+        .mapIndexed { index, s -> index + 1 to coverageMarkerRegex.find(s) }
+        .filter { it.second != null }
+        .toMap()
+        .mapValues { it.value!!.groupValues[1] }
+
+internal fun pathToFile(vararg names: String) = File(names.joinToString(File.separator))
```

Comment:
it may be slightly better to use Paths.get("src", ... the rest... ).toFile() to avoid direct use of File.separator